pyqtgraph/functions.py

```diff
@@ -1752,51 +1735,120 @@ def arrayToQPath(x, y, connect='all'):
             if first < len(x):
                 # Replace all non-finite entries from beginning of arr with the first finite one
                 idx[:first] = first
-                arr[1:-1] = arr[1:-1][idx]
+                arr[:] = arr[:][idx]
 
     # decide which points are connected by lines
     if eq(connect, 'all'):
-        arr[1:-1]['c'] = 1
+        polygon = arrayToQPolygonF(
+            arr['x'],
+            arr['y']
+        )
+        path.addPolygon(polygon)
+        return path
     elif eq(connect, 'pairs'):
-        arr[1:-1]['c'][::2] = 0
-        arr[1:-1]['c'][1::2] = 1  # connect every 2nd point to every 1st one
+        if qt6:
+            points = np.insert(arr[['x', 'y']], np.arange(2, len(x), step=2, dtype=int), np.nan, axis=0)
+            polygon = arrayToQPolygonF(points['x'], points['y'])
+            path.addPolygon(polygon)
+            return path
+        else:
+            arr[:]['c'][::2] = 0
+            arr[:]['c'][1::2] = 1  # connect every 2nd point to every 1st one
     elif eq(connect, 'finite'):
         # Let's call a point with either x or y being nan is an invalid point.
         # A point will anyway not connect to an invalid point regardless of the
         # 'c' value of the invalid point. Therefore, we should set 'c' to 0 for
         # the next point of an invalid point.
-        if isfinite is None:
+        if qt6:
+            # look for the first finite element.
             isfinite = np.isfinite(x) & np.isfinite(y)
-        arr[2:]['c'] = isfinite
+            argNaNs = np.argwhere(isfinite)
+            first = 0 if len(argNaNs) == 0 else argNaNs[0]
```

Comment:
Should be
```python
first = 0 if len(argNaNs) == 0 else argNaNs.item(0)
```
in order to convert into a Python scalar, otherwise the slicing fails

Reply:
nice catch, I would have pulled my hair out with this one.